components/p3a/brave_p3a_rotation_scheduler.h

```diff
@@ -0,0 +1,66 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_P3A_BRAVE_P3A_ROTATION_SCHEDULER_H_
+#define BRAVE_COMPONENTS_P3A_BRAVE_P3A_ROTATION_SCHEDULER_H_
+
+#include <memory>
+
+#include "base/callback.h"
+#include "base/containers/flat_map.h"
+#include "base/time/time.h"
+#include "base/timer/wall_clock_timer.h"
+#include "brave/components/p3a/brave_p3a_config.h"
+#include "brave/components/p3a/metric_log_type.h"
+
+class PrefService;
+class PrefRegistrySimple;
+
+namespace brave {
+
+class BraveP3ARotationScheduler {
```

Comment:
nice to finally have it extracted from the service

Reply:
yes agreed, thanks